extensions/gmsa-dc/v1/Setup-gMSA.ps1

```diff
@@ -0,0 +1,50 @@
+﻿##################################################################################
+# This script will automate the creation of a Group Managed Service Account,
+# install the account locally, produce the JSON file with gMSA info, and convert
+# it to the Yaml file needed.
+#
+# This is only for automated e2e testing.  DO NOT use this for production.
+# Jeremy Wood (JeremyWx)
+# Version: 1.0.0.0
+##################################################################################
+
+# Change to working directory
+Set-Location -Path C:\gmsa
+
+# Script Logging - Feel free to comment out if you like.
+Start-Transcript -Path "C:\gmsa\Setup-gmsa.txt" -Append
+# Making sure the AD services are up and running
+Start-Sleep -Seconds 60
+Import-Module ActiveDirectory
+
+if (Get-AdGroupMember -Identity "Enterprise Admins" | Select-String -Pattern "gmsa-admin" -Quiet) {
+    
+	# Add the KDS Root Key
+    $KdsRootKey = Get-KdsRootKey
+    if ($null -eq $KdsRootKey.KeyId) {
+        Add-KdsRootKey –EffectiveTime ((get-date).addhours(-10))

```

Comment:
A comment explaining this would be nice to future maintainers :)

(I remember reading something about it being the max propagation delay throughout a cluster, or something?)